### PR TITLE
feat(SampleDistanceSlider): Rewrite in bootstrap and optimize

### PR DESCRIPTION
**Pull request by labdala**
- Create slider using Form component;
- Use `div` to introduce unclickable image;
- Optimize by using `useSelector` and requesting just the needed state component prop, and by applying React.memo

Co-authored-by: Paul Elliott <paul.elliott@kitware.com>

### CODE REVIEW

File: src/App.scss [hunk 1]
```diff
@@ -20,3 +20,22 @@
...
+  
+  &.checked {
+    background-color: transparent !important;
+    border-color: transparent !important;
+  }
+
+  & > img {
+    width: 1.2rem;
+  }
+}
```
Comment (PaulHax) [inline src/App.scss:41]: Good to simplify this as just an image, not a button.  If it is just a button, can probably get away with just the `width` property and remove the rest?  See if we can simplify this.

Also, can you fix my mistake and move these 2 SCSS rules to the big CSS file?  We will just use SCSS to compile the pyData theme.

File: src/Images/SampleDistanceSlider.jsx [hunk 2]
```diff
@@ -1,66 +1,73 @@
...
-        onChange={(_e, val) => {
-          spacingChanged(val)
-        }}
-      />
+        <div className="icon-button-disabled" ref={spacingDiv}>
+          <Image src={sampleDistanceIconDataUri}></Image>
+        </div>
+      </OverlayTrigger>
+      <Form>
+        <Form.Group>
```
Comment (PaulHax) [inline src/Images/SampleDistanceSlider.jsx:53]: I'm not sure, but we may not need `Form.Group` if we only have one `Form.Control`

File: src/Images/SampleDistanceSlider.jsx [hunk 3]
```diff
@@ -1,66 +1,73 @@
...
 
-function SampleDistanceSlider(props) {
+const SampleDistanceSlider = React.memo(function SampleDistanceSlider(props) {
   const { service } = props
   const spacingDiv = useRef(null)
   const spacingElement = useRef(null)
-  const [state, send] = useActor(service)
-  const name = state.context.images.selectedName
-  const actorContext = state.context.images.actorContext.get(name)
+  const stateContext = useSelector(service, (state) => state.context)
```
Comment (PaulHax) [inline src/Images/SampleDistanceSlider.jsx:17]: I spy with my little eye...

File: src/Images/SampleDistanceSlider.jsx [hunk 4]
```diff
@@ -1,66 +1,73 @@
...
+import Button from 'react-bootstrap/Button'
+import Form from 'react-bootstrap/Form'
+import Image from 'react-bootstrap/Image'
+import Tooltip from 'react-bootstrap/Tooltip'
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
+import cn from 'classnames'
 
-function SampleDistanceSlider(props) {
+const SampleDistanceSlider = React.memo(function SampleDistanceSlider(props) {
   const { service } = props
```
Comment (PaulHax) [inline src/Images/SampleDistanceSlider.jsx:14]: We should not need `React.memo` if the only prop we are passing is `service` which should not change over the lifetime of the app.  After we get these PRs merged, maybe we look up the component tree and tackle the useActors at the top levels and stop the top components from re-rendering on each state change.  If they stop rendering all the time, then we probably don't need React.memo on all of our little components.

File: src/Images/SampleDistanceSlider.jsx [hunk 5]
```diff
@@ -1,66 +1,73 @@
...
+      <Form>
+        <Form.Group>
+          <Form.Control
+            ref={spacingElement}
+            type="range"
+            custom
+            className="slider"
+            min={0}
+            max={1}
+            value={actorContext.volumeSampleDistance}
```
Comment (PaulHax) [inline src/Images/SampleDistanceSlider.jsx:61]: When I drag the slider, the sample distance changes, but the tick mark does not move.  Something is up here...

**Review by PaulHax — CHANGES_REQUESTED** [review]

File: src/Images/SampleDistanceSlider.jsx (hunk 4 above)
Comment (labdala) [reply src/Images/SampleDistanceSlider.jsx:14]: Got it! So it makes sense to not worry about the re-renderings now and tackle that in the next step.

File: src/Images/SampleDistanceSlider.jsx (hunk 2 above)
Comment (labdala) [reply src/Images/SampleDistanceSlider.jsx:53]: You are right! Thanks for pointing that out.

File: src/App.scss (hunk 1 above)
Comment (labdala) [reply src/App.scss:41]: Yes, I am on it.

File: src/Images/SampleDistanceSlider.jsx (hunk 5 above)
Comment (PaulHax) [reply src/Images/SampleDistanceSlider.jsx:61]: What the what!  Some sort of re-render update order thing???  Flow should be `slider event -> send state event -> state system magic -> useSelector triggers re-render with updated volumeSampleDistance`.

Comment (labdala) [reply src/Images/SampleDistanceSlider.jsx]: [Someone had a similar issue](https://stackoverflow.com/questions/64882133/react-memo-not-re-render-if-props-change), but I am not sure if what is happening there applies to our case. I suspect that the issue is in the `spacingChanged` -- it gets that the value is changing as we toggle the slider, but I guess it does not send that change to the parent when I use React.memo?!

In any case, I deleted the React.memo in the next versions for the reasons you pointed out above.

Comment (PaulHax) [reply src/Images/SampleDistanceSlider.jsx]: OK, I'll update performance issue to track this.

Comment (labdala) [reply src/Images/SampleDistanceSlider.jsx]: Thanks!

File: src/Images/SampleDistanceSlider.jsx (hunk 3 above)
Comment (labdala) [reply src/Images/SampleDistanceSlider.jsx:15]: Oh 💡 
I can use multiple selectors here as you suggested [here](https://github.com/InsightSoftwareConsortium/itk-viewer-bootstrap-ui/pull/7/files#r888410436) .

**Review by PaulHax — APPROVED** [review]
Much cleaner.

**Pull request merged**